api/client/service/inspect.go

```diff
@@ -114,23 +113,27 @@ func printService(out io.Writer, service swarm.Service) {
 	fmt.Fprintf(out, "ContainerSpec:\n")
 	printContainerSpec(out, service.Spec.TaskTemplate.ContainerSpec)
 
-	if service.Spec.TaskTemplate.Resources != nil {
+	resources := service.Spec.TaskTemplate.Resources
+	if resources != nil {
 		fmt.Fprintln(out, "Resources:")
 		printResources := func(out io.Writer, r *swarm.Resources) {
 			if r.NanoCPUs != 0 {
-				fmt.Fprintf(out, " CPU:\t\t%g\n", float64(r.NanoCPUs)/1e9)
+				fmt.Fprintf(out, "  CPU:\t\t%g\n", float64(r.NanoCPUs)/1e9)
 			}
 			if r.MemoryBytes != 0 {
-				fmt.Fprintf(out, " Memory:\t\t%s\n", units.BytesSize(float64(r.MemoryBytes)))
+				fmt.Fprintf(out, "  Memory:\t\t%s\n", units.BytesSize(float64(r.MemoryBytes)))
 			}
 		}
-		if service.Spec.TaskTemplate.Resources.Reservations != nil {
-			fmt.Fprintln(out, "Reservations:")
-			printResources(out, service.Spec.TaskTemplate.Resources.Reservations)
+
+		if resources.Reservations != nil &&
+			(resources.Reservations.NanoCPUs != 0 || resources.Reservations.MemoryBytes != 0) {
+			fmt.Fprintln(out, " Reservations:")
```

Comment:
Cosmetic suggestion: fold this stuff into `printResources` and add an argument for the heading. Then this part of the code becomes

```go
printResources(out, "Reservations", resources.Reservations)
printResources(out, "Limits", resources.Limits)
```

I think it will make the code simpler, especially if we add more resource types in the future.